reframe/schemas/runreport.json

```diff
@@ -0,0 +1,53 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/eth-cscs/reframe/master/schemas/runreport.json",
+    "title": "Validation schema for ReFrame's run report",
+    "type": "array",
+    "items": {
+        "type": "object",
+        "properties": {
+            "testname": {"type": "string"},
+            "description": {"type": ["string"]},
+            "system": {"type": ["string", "null"]},
```

Comment:
System can't be null.

Reply:
I think we need to allow null for cases like `BadSetupCheckEarly` of `unittests/resources/checks/frontend_checks.py`, that raise an exception after the setup. There ,`check.current_partition` is `None`.

Reply:
Yes, but `current_system` is never `None`. I think that you set the `system` first to the value of `current_system` instead of `None`.